src/main/scala/nodes/images/Convolver.scala

```diff
@@ -101,29 +101,31 @@ object Convolver {
       whitener: Option[ZCAWhitener],
       varConstant: Double): DenseMatrix[Double] = {
     var x,y,chan,pox,poy,py,px = 0
-    while (x < resWidth) {
-      y = 0
-      while (y < resHeight) {
-        chan = 0
-        while (chan < imgChannels) {
-          poy = 0
-          while (poy < convSize) {
-            pox = 0
-            while (pox < convSize) {
+
+    poy = 0
+    while (poy < convSize) {
```

Comment:
What I mean is - I want something that converts somehting like
```
specialFor(
  a <- 0 until amax;
  b <- 0 until bmax;
  c <- 0 until cmax;
  d <- 0 until dmax;
  e <- 0 until emax) {

//do something with a,b,c,d,e
}
```

Into a loop like this at compile time.